lib/game.rb

```diff
@@ -138,14 +139,8 @@ def move(player)
 		symbol = player.symbol
 		puts "#{player.name} move (print a number from 0 to #{board.dim-1} and a letter from A to #{ALPHABET[board.dim-1]}) or 'Q' to quit\n"
 		player_input = gets.gsub(/\s+/, "").split('')
-		if board.valid_input?(player_input)
-			move_index = board.dim*(player_input[0].to_i)+ALPHABET.index(player_input[1].upcase)
-			if board.empty_space?(move_index)
-				board.change_matrix(player.symbol, move_index)
-			else
-				puts 'Invalid move'
-				move(player)
-			end
+		if board.valid_move?(player_input)
+			board.change_matrix(player.symbol, board.move_index(player_input))
 		elsif player_input.length == 1 && player_input[0].upcase == 'Q'
 			@wanna_quit = true
 		else
```

Comment:
I don't think it is the responsibility of the `board` object to validate the input. 
Can we split this method? The actions that happen here are:
1. You get input from the user through the console
2. You validate the input
3. If the input is valid you send it to the board, else you request it again
4. You 'write' to the board